### PR TITLE
Add note about using choice_label as a callback

### PR DESCRIPTION
**Pull request by mhlavac**
I was wondering how to call a method with an argument for some time and then I've came with an idea of using ArrayAccess that can be used with PropertyAccessor.

Then I found I can actually use callback in `choice_label`, but this was not documented. Is this correct usage?

### CODE REVIEW

**Comment by stof** [comment]
yes, it is a supported use case (it is even the main API, as property path are transformed into a callable internally)

**Comment by mhlavac** [comment]
Ok, is the example ok, or should I also add index in function args so it's clear to everyone?

**Comment by stof** [comment]
IMO, you need to document the arguments received by the callable (even though most cases will only need the first one).

You should also change the type of the option, to say ``string|callable``.

And I'm not sure this should be a note actually. It should be documented at the same level of importance than the property path usage IMO. But I will let the doc team make the choice here.

**Comment by mhlavac** [comment]
@stof thank you. I've updated it, if they accept it then I'll squash the commits into one.

**Comment by javiereguiluz** [comment]
@mhlavac there is no need to squash the commits because `symfony` and `symfony-docs` repos are managed with a tool that does the squashing automatically.

File: reference/forms/types/entity.rst [hunk 1]
```diff
@@ -110,12 +110,22 @@ choice_label
...
 as text in the HTML element. If left blank, the entity object will be
 cast into a string and so must have a ``__toString()`` method.
 
+The ``choice_label`` option can also be a callback that accepts 2 arguments: entity and index.
+You can use it like this::
+    
+      $builder->add('gender', 'entity', array(
+         'class' => 'MyBundle:Gender',
+         'choice_label' => function(MyBundle\Entity\Gender $gender, $index) {
+           return $index . '. ' . $gender->getLocalizedGender($this->locale);
```
Comment (OskarStark) [inline reference/forms/types/entity.rst]: indention correct here?

**Comment by xabbuh** [comment]
First, thank you very much for your contribution @mhlavac. Unfortunately, someone already opened a similar PR (see #5755). Also we have a lot more possibilities to configure the label path and most of this also applies to some other options (see my comment in https://github.com/symfony/symfony-docs/pull/5755#issuecomment-146308077). Maybe you and @aivus would like to work together on this?

**Comment by HeahDude** [comment]
Should be closed as fixed by #5876

**Pull request closed without merging**